connectorx/src/transports/postgres_arrow.rs

```diff
@@ -9,24 +8,23 @@ pub struct PostgresArrowTransport;
 
 impl_transport!(
     name = PostgresArrowTransport,
-    systems = PostgresTypeSystem => DummyTypeSystem,
+    systems = PostgresTypeSystem => ArrowTypeSystem,
     route = PostgresSource<Binary> => ArrowDestination,
     mappings = {
-        { Float4[f32]                => F64[f64]                | conversion all }
-        { Float8[f64]                => F64[f64]                | conversion all }
-        { Int2[i16]                  => I64[i64]                | conversion all }
-        { Int4[i32]                  => I64[i64]                | conversion all }
-        { Int8[i64]                  => I64[i64]                | conversion all }
-        { Bool[bool]                 => Bool[bool]              | conversion all  }
-        { Text[&'r str]              => String[String]          | conversion half }
-        { BpChar[&'r str]            => String[String]          | conversion none }
-        { VarChar[&'r str]           => String[String]          | conversion none }
-        { Timestamp[NaiveDateTime]   => DateTime[DateTime<Utc>] | conversion half }
-        { TimestampTz[DateTime<Utc>] => DateTime[DateTime<Utc>] | conversion all }
-        { Date[NaiveDate]            => DateTime[DateTime<Utc>] | conversion half }
-        { UUID[Uuid]                 => String[String]          | conversion half }
-        { Char[&'r str]              => String[String]          | conversion none}
-        // { Time[NaiveTime]            => String[String]          | conversion half }
+        { Float4[f32]                => Float32[f32]            | conversion all }
+        { Float8[f64]                => Float64[f64]            | conversion all }
+        { Int2[i16]                  => Int32[i32]              | conversion all }
+        { Int4[i32]                  => Int32[i32]              | conversion all }
+        { Int8[i64]                  => Int64[i64]              | conversion all }
+        { Bool[bool]                 => Boolean[bool]           | conversion all  }
```

Comment:
There are three conversion rule generating options: all, half and none.

* `all` means the `TypeConversion<T,U>` can be done trivially using rust's `as`, e.g. `f32` to `f32` or `f32` to `f64`.
* `half` means we'd like to manually implement the `TypeConversion<T,U>` for non-optional type, e.g. `i64` to `Date64` and auto generate the type conversion of `TypeConversion<Option<T>,Option<U>>` by  `t.map(TypeConversion::convert)`.
* `none` means we'd like to manually implement the `TypeConversion<T,U>` and `TypeConversion<Option<T>, Option<U>>`, which should be rarely used.

Reply:
Thanks for the explanation. :+1: Makes more sense now.